nixos/modules/services/security/crowdsec.nix

```diff
@@ -489,182 +398,189 @@ in
-        }"
-      ]
-      ++ lib.optionals (cfg.hub.appSecRules != [ ]) [
-        "${lib.getExe cscli} appsec-rules install ${
-          lib.strings.concatMapStringsSep " " (x: lib.escapeShellArg x) cfg.hub.appSecRules
-        }"
-      ]
-      ++ lib.optionals (cfg.settings.general.api.server.enable) [
-        ''
-          if [ ! -s "${cfg.settings.general.api.client.credentials_path}" ]; then
-            ${lib.getExe cscli} machine add "${cfg.name}" --auto
-          fi
-        ''
-      ]
-      ++ lib.optionals (cfg.settings.capi.credentialsFile != null) [
-        ''
-          if ! ${lib.getExe pkgs.gnugrep} -q password "${cfg.settings.capi.credentialsFile}" ]; then
-            ${lib.getExe cscli} capi register
-          fi
-        ''
-      ]
-      ++ lib.optionals (cfg.settings.console.tokenFile != null) [
-        ''
-          if [ ! -e "${cfg.settings.console.tokenFile}" ]; then
-            ${lib.getExe cscli} console enroll "$(cat ${cfg.settings.console.tokenFile})" --name ${cfg.name}
-          fi
-        ''
-      ];
+      ++ lib.optional cfg.settings.general.api.server.enable ''
+        if [ ! -f ${cfg.settings.general.api.client.credentials_path} ]; then
+          echo "No local API credentials currently created. Generating local API credentials..."
+          ${lib.getExe cscli} machines add nixos-generated --auto --file ${cfg.settings.general.api.client.credentials_path}
+        fi
+      ''
+      ++
+        lib.optional (cfg.hub.collections != [ ])
+          "${lib.getExe cscli} collections install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.collections
+          }"
+      ++
+        lib.optional (cfg.hub.scenarios != [ ])
+          "${lib.getExe cscli} scenarios install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.scenarios
+          }"
+      ++
+        lib.optional (cfg.hub.parsers != [ ])
+          "${lib.getExe cscli} parsers install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.parsers
+          }"
+      ++
+        lib.optional (cfg.hub.postOverflows != [ ])
+          "${lib.getExe cscli} postoverflows install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.postOverflows
+          }"
+      ++
+        lib.optional (cfg.hub.appSecConfigs != [ ])
+          "${lib.getExe cscli} appsec-configs install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.appSecConfigs
+          }"
+      ++
+        lib.optional (cfg.hub.appSecRules != [ ])
+          "${lib.getExe cscli} appsec-rules install ${
+            lib.strings.concatMapStringsSep " " lib.escapeShellArg cfg.hub.appSecRules
+          }"
+      ++ lib.optional (cfg.settings.general.api.server.online_client.credentials_path != null) ''
+        if ! grep -q password "${cfg.settings.general.api.server.online_client.credentials_path}";  then
```

Comment:
Is the `grep` found from PATH or should it read `${lib.getExe' pkgs.gnugrep grep}` instead?